docs/madr/decisions/009-tracing-policy.md

```diff
@@ -0,0 +1,184 @@
+- `MeshTrafficTrace`
+
+#### Backends
+
+Currently, tracing support requires defining _backends_ in the `Mesh` resource.
+A backend defines where traces are sent, the _provider_. For `TrafficTrace`, it also contains the sampling rates settings.
+The `TrafficTrace` resource references these backends and contains no configuration of its own.
+
+The proposed `Tracing` policy follows the decisions about backends as laid out in
+[the MADR for `MeshAccessLog`](docs/madr/decisions/009-tracing-policy.md#backends). This
+means creating a new `TracingBackend` resource and allowing inline backend
+definitions.
+
+However, we propose moving configuring sampling and tag configuration
+from the backend (the `TrafficTrace` status quo) to the `Tracing` resource.
+Backends can contain the same settings but they can be set or overriden in
+specific `Tracing` resources.
+
+#### `targetRef`
+
+This is a new policy so it's based on `targetRef`. Envoy tracing is configured
+on the HTTP connection manager so `Tracing` has a single `spec.targetRef` field
+that selects what it applies to. It does not use `to` or `from` fields.
+
+All logging configuration happens under `spec.default` so that users are able to
+override settings with more specific `targetRef`s.
+
+Resources supported by `spec.targetRef` are `Mesh`, `MeshSubset`, `MeshService`,
+`MeshServiceSubset`, `MeshGatewayRoute` and eventually the future
+evolution of `TrafficRoute`, which we'll call `MeshTrafficRoute` in this
+document.
+
+#### Policy-specific configuration
+
+This MADR proposes additional options be configurable on `Tracing` resources
+directly.
+
+##### Sampling
+
+At the moment, tracing backends only support Envoy's
+[`overall_sampling`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-tracing-overall-sampling)
+via the `sampling` field.
+
+This MADR proposes instead the following:
+
+```yaml
+spec:
+ default:
+  sampling:
+   overall: <percentage>
+   client: <percentage>
+   random: <percentage>
+```
+
+with the justification that it's not clear to users
+that when they set the simple `sampling` field we have now,
+whether traces started via `x-client-trace-id` are limited.
+This also enables disabling `x-client-trace-id`.
+
+##### Tags
+
+Envoy can set tags in traces using values from:
+
+- Literal value
+- Request header
+- Environment variable
+- Metadata
+
+We propose allowing users to configure custom tags using only literal values or
+request headers. Using Kuma leaves environment variables and Envoy metadata
+opaque for users so the use cases for configuring them (at least directly)
+in Kuma are limited.
+
+Tags can be configured as follows:
+
+```yaml
+spec:
+ default:
+  tags:
+   - name:
+     # exactly one of the following keys must be set
+     literal: ...
+     header:
+      name:
+      default:
+```
```

Comment:
what about kuma tags from DP object? Do we already send kuma tags automatically?

Reply:
Not as far as I can tell.

Reply:
can we add this then?

Reply:
I think this MADR leaves the door open for it. Do you think it needs to be in this MADR?

Reply:
Add a note about DP object potentially possible later